webapp/channels/src/components/permissions_gates/team_permission_gate/team_permission_gate.tsx

```diff
@@ -23,26 +23,19 @@ type Props = {
     /**
      * Invert the permission (used for else)
      */
-    invert: boolean;
+    invert?: boolean;
 
     /**
      * Content protected by the permissions gate
      */
     children: React.ReactNode;
 };
 
-export default class TeamPermissionGate extends React.PureComponent<Props> {
-    public static defaultProps = {
-        invert: false,
-    };
-
-    render() {
-        if (this.props.hasPermission && !this.props.invert) {
-            return this.props.children;
-        }
-        if (!this.props.hasPermission && this.props.invert) {
-            return this.props.children;
-        }
-        return null;
+const TeamPermissionGate = ({invert = false, hasPermission, children}: Props) => {
+    if (hasPermission !== invert) {
+        return <>{children}</>;
     }
-}
+    return null;
+};
+
+export default React.memo(TeamPermissionGate);
```

Comment:
@larkox this line is causing the issue if we dont wrap it in a memo function it works as expected.

Reply:
@hmhealey or @M-ZubairAhmed  Any idea why this may be happening and how to avoid it? The snapshots look wrong, but the code seems to be doing what it should.

Reply:
In those snapshots that changed, something is causing the component to render its children when it didn't previously. I think the difference might be because the logic inside the component has changed subtly because `!==` is stricter about typing than the old code was

Reply:
Actually, no, that's not the case. I tried using the old logic and the snapshots still match what they are now. I think the difference is actually in how Enzyme renders components when you use `mount` because the snapshot contains React elements and not plain HTML like I would've expected. The code in the tests still appears to be working correctly to me, but the snapshots aren't helpful any more because they always render the text inside of the `TeamPermissionGate`.

Because of that, I think the tests for this component need to be updated to not use snapshots and instead check to see if the text is actually rendered. Instead of `expect(wrapper).toMatchSnapshot()`, you'll want them to use `expect(wrapper.text()).toContain(...)` when the children should be rendered and `expect(wrapper.text()).not.toContain(...)` when they shouldn't be.

Alternatively, if you want bonus points, we're wanting to migrate all of our tests to React Testing Library instead of Enzyme, and it handles this sort of testing better, but that's not necessary if you don't want to. In that case, you'd write it like `expect(screen.findByText(...)).toBeVisible()`

Reply:
@hmhealey should I update all the test for this file or only the ones that don't match the previous snapshot?